subsys/bluetooth/audio/cap_commander.c

```diff
@@ -77,17 +81,240 @@ int bt_cap_commander_discover(struct bt_conn *conn)
+		add_src_param.broadcast_id = proc_param->broadcast_reception_start.broadcast_id;
+		add_src_param.pa_interval = proc_param->broadcast_reception_start.pa_interval;
+		add_src_param.num_subgroups = proc_param->broadcast_reception_start.num_subgroups;
+		add_src_param.subgroups = proc_param->broadcast_reception_start.subgroups;
+
+		active_proc->proc_initiated_cnt++;
+		err = bt_bap_broadcast_assistant_add_src(conn, &add_src_param);
+
+		if (err != 0) {
+			LOG_DBG("Failed to perform broadcast reception start for conn %p: %d",
+				(void *)conn, err);
+			bt_cap_common_abort_proc(conn, err);
+			cap_commander_unicast_audio_proc_complete();
+		}
+
+	} else {
+		cap_commander_unicast_audio_proc_complete();
+	}
+}
+
+static int cap_commander_register_ba_cb(void)
+{
+	int err = 0;
+
+	bt_bap_broadcast_assistant_register_cb(&broadcast_assistant_cb);
+	if (err != 0) {
+		LOG_DBG("Failed to register broadcast assistant callbacks: %d", err);
+
+		return -ENOEXEC;
+	}
+
+	ba_cb_registered = true;
+
+	return 0;
+}
+
+static bool valid_broadcast_reception_start_param(
+	const struct bt_cap_commander_broadcast_reception_start_param *param)
+{
+	CHECKIF(param == NULL) {
+		LOG_DBG("param is NULL");
+		return false;
+	}
+
+	CHECKIF(param->count == 0) {
+		LOG_DBG("Invalid param->count: %u", param->count);
+		return false;
+	}
+
+	CHECKIF(param->count > CONFIG_BT_MAX_CONN) {
+		LOG_DBG("param->count (%zu) is larger than CONFIG_BT_MAX_CONN (%d)", param->count,
+			CONFIG_BT_MAX_CONN);
+		return false;
+	}
+
+	CHECKIF(param->param == NULL) {
+		LOG_DBG("param->param is NULL");
+		return false;
+	}
+
+	for (size_t i = 0; i < param->count; i++) {
+		const struct bt_cap_commander_broadcast_reception_start_member_param *start_param =
+			&param->param[i];
+		const union bt_cap_set_member *member = &param->param[i].member;
+		const struct bt_cap_common_client *client =
+			bt_cap_common_get_client(param->type, member);
+
+		if (client == NULL) {
+			LOG_DBG("Invalid param->param[%zu].member", i);
+			return false;
+		}
+
+		for (size_t j = 0U; j < i; j++) {
+			const union bt_cap_set_member *other = &param->param[j].member;
+
+			if (other == member) {
+				LOG_DBG("param->members[%zu] (%p) is duplicated by "
+					"param->members[%zu] (%p)",
+					j, other, i, member);
+				return false;
+			}
+		}
+
+		CHECKIF(start_param->num_subgroups == 0) {
+			LOG_DBG("param->param[%d]->num_subgroups is 0", i);
+			return false;
+		}
+
+		CHECKIF(start_param->subgroups == NULL) {
+			LOG_DBG("param->param[%d]->subgroup is NULL", i);
+			return false;
+		}
+	}
+
+	return true;
+}
+
 int bt_cap_commander_broadcast_reception_start(
 	const struct bt_cap_commander_broadcast_reception_start_param *param)
 {
-	return -ENOSYS;
+	const struct bt_cap_commander_proc_param *proc_param;
+	struct bt_cap_common_proc *active_proc;
+	struct bt_conn *conn;
+	struct bt_bap_broadcast_assistant_add_src_param add_src_param = {0};
+	int err;
```

Comment:
It's not a requirement in our coding guidelines, but I like to (at the extend possible) to order parameters in reverse Christmas tree fashion:

 ```suggestion
	struct bt_bap_broadcast_assistant_add_src_param add_src_param = {0};
	const struct bt_cap_commander_proc_param *proc_param;
	struct bt_cap_common_proc *active_proc;
	struct bt_conn *conn;
	int err;
```